fix parsing of port_http in combination with spaces

lobeck opened the pull request:
- fixes braice/MuMuDVB#102
- previously the setting only properly worked when used without any spaces between the setting and the equals sign (port_http=)
- this is now fixed and an arbitrary number of spaces may be used (like any other setting)

lobeck committed:
fix parsing of port_http in combination with spaces

- fixes braice/MuMuDVB#102
- previously the setting only properly worked when used without any spaces between the setting and the equals sign (port_http=)
- this is now fixed and an arbitrary number of spaces may be used (like any other setting)

braice commented:
Hello

can you give a try to put spaces in the equation and see if it still works ?

Thx

2015-10-29 11:18 GMT+01:00 lobeck notifications@github.com:

> - fixes braice/MuMuDVB#102
>   https://github.com/braice/MuMuDVB/issues/102
> - previously the setting only properly worked when used without any
>   spaces between the setting and the equals sign (port_http=)
> - this is now fixed and an arbitrary number of spaces may be used
>   (like any other setting)
> 
> ---
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/braice/MuMuDVB/pull/104
> Commit Summary
> - fix parsing of port_http in combination with spaces
> 
> File Changes
> - _M_ src/unicast_http.c
>   https://github.com/braice/MuMuDVB/pull/104/files#diff-0 (2)
> 
> Patch Links:
> - https://github.com/braice/MuMuDVB/pull/104.patch
> - https://github.com/braice/MuMuDVB/pull/104.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/braice/MuMuDVB/pull/104.

lobeck commented:
Hi,

yes, i'm already using the check since i created the pull request and i'm using the following line:

```
  port_http            = 2000+%card
```

Works like a charm.

Cheers!

lobeck commented:
Note: i have updated the comment above several times due to formatting issues with the amount of spaces i have in my config. The above example now is the one i have with 4 spaces in the beginning (due to config management) and 13 spaces between `port_http` and the equals sign.

braice commented:
One final check : around the plus sign ?

2015-10-30 15:10 GMT+01:00 lobeck notifications@github.com:

> Note: i have updated the comment above several times due to formatting
> issues with the amount of spaces i have in my config. The above example now
> is the one i have with 4 spaces in the beginning (due to config management)
> and 13 spaces between port_http and the equals sign.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/braice/MuMuDVB/pull/104#issuecomment-152536525.

lobeck commented:
tested this:

```
  port_http            = 2000 + %card
```

does not work with the patch, but is it supposed to work that way?

If i do for example:

```
  port_http            = 2010 + %card
```

It binds to port 2010 even if the card is 1, so the setting itself is still working but either this is not saved correctly in `portOut_str` or spaces are not supported by `string_comput`.

m0yellow commented:
On 02 Nov 2015, at 17:42, lobeck notifications@github.com wrote:

> tested this:
> 
>   port_http            = 2000 + %card
> 
> does not work with the patch, but is it supposed to work that way?
> I would say this is supposed to work. If not we would need to clarify that in the documentation.

braice commented:
Le 2 nov. 2015 5:42 PM, "lobeck" notifications@github.com a écrit :

> tested this:
> 
>   port_http            = 2000 + %card
> 
> does not work with the patch, but is it supposed to work that way?

Yes it should but I don't know what is broken

Can you print the extracted string?  If it contains only a part of the
equation can you try with "="  instead of delimiteurs

Best

lobeck committed:
remove spaces as delimiter for port_http

if there are also spaces in the delimiter, this causes a parsing error
if there is a formula following which is also containing spaces

lobeck commented:
Your guess was right and the extracted string was just `2000`. This is caused by `strtok`, which is not using the delimiter as a sequence but as a set of possible delimiters.

So i've now updated the fork to just use the equals sign as delimiter, that way i managed to create a working testcase:

```
port_http = 2000 + %card
token1: port_http
token2:  2000 + %card
replaced:  2000 + 10
computed: 2010
```

Note that `token2` is preceeded by a space, but `mumu_string_replace` and `string_comput` can handle this (as you see in the output above).

For reference you'll find the test i used here: https://gist.github.com/lobeck/627d8bd5735f29d7a630#file-port_http_test-c-L121-L147
I'm sorry if it's a bit ugly, but this is the first time i'm messing with C and i just tried to get it working quickly.

lobeck committed:
rework port_http parsing

- the current implementation had some issues with spaces in general at various places across the setting
- now all spaces are stripped and the usual parsing can continue
- other settings might be affected as well, but i haven't touched them as i don't know the impact

lobeck commented:
reworked the whole patch, das there have still been issues if there is more than once space before the equals sign. I've now modified it, to strip all spaces and if this produces an empty string, do another round of tokenizing to get the actual setting. It's also tested if there are multiple spaces _after_ the equals sign.

Here is the updated test as well: https://gist.github.com/lobeck/627d8bd5735f29d7a630#file-port_http_test-c-L141-L158

braice commented:
Hi Can you make a new pull request for your new patch ?

Thank you

2015-11-03 11:33 GMT-05:00 lobeck notifications@github.com:

> reworked the whole patch, das there have still been issues if there is
> more than once space before the equals sign. I've now modified it, to strip
> all spaces and if this produces an empty string, do another round of
> tokenizing to get the actual setting. It's also tested if there are
> multiple spaces _after_ the equals sign.
> 
> Here is the updated test as well:
> https://gist.github.com/lobeck/627d8bd5735f29d7a630#file-port_http_test-c-L141-L158
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/braice/MuMuDVB/pull/104#issuecomment-153408295.

lobeck commented:
Hi,

sorry, i've just now seen your reply.

What do you mean by a new pull request? Everything i changed after i created this pull request is automatically added. If you look over my last comment, you'll see the last commit i did on this (pull requests are automatically updated if a commit is pushed).

If you want this for a fresh discussion, it's ok, then i'll just close this and open a new one with clean comments on it.

braice commented:
Hello

Yes please start a fresh discussion with a new request for your commits, I
am a bit confused by what guthub did on this one

Thank you

2016-03-31 7:14 GMT-04:00 Christian Becker notifications@github.com:

> Hi,
> 
> sorry, i've just now seen your reply.
> 
> What do you mean by a new pull request? Everything i changed after i
> created this pull request is automatically added. If you look over my last
> comment, you'll see the last commit i did on this (pull requests are
> automatically updated if a commit is pushed).
> 
> If you want this for a fresh discussion, it's ok, then i'll just close
> this and open a new one with clean comments on it.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> https://github.com/braice/MuMuDVB/pull/104#issuecomment-203883030
